packages/osd-test/src/osd/osd_test_config.ts

```diff
@@ -48,32 +48,43 @@ export const osdTestConfig = new (class OsdTestConfig {
   getUrlParts(): UrlParts {
     // allow setting one complete TEST_OPENSEARCH_DASHBOARDS_URL for opensearch like https://opensearch:changeme@example.com:9200
     if (process.env.TEST_OPENSEARCH_DASHBOARDS_URL) {
-      const testOpenSearchDashboardsUrl = url.parse(process.env.TEST_OPENSEARCH_DASHBOARDS_URL);
+      const testOpenSearchDashboardsUrl = new URL('', process.env.TEST_OPENSEARCH_DASHBOARDS_URL);
+      testOpenSearchDashboardsUrl.pathname = '';
       return {
         protocol: testOpenSearchDashboardsUrl.protocol?.slice(0, -1),
         hostname: testOpenSearchDashboardsUrl.hostname ?? undefined,
         port: testOpenSearchDashboardsUrl.port
           ? parseInt(testOpenSearchDashboardsUrl.port, 10)
           : undefined,
-        auth: testOpenSearchDashboardsUrl.auth ?? undefined,
-        username: testOpenSearchDashboardsUrl.auth?.split(':')[0],
-        password: testOpenSearchDashboardsUrl.auth?.split(':')[1],
+        auth:
+          testOpenSearchDashboardsUrl.username && testOpenSearchDashboardsUrl.password
+            ? `${testOpenSearchDashboardsUrl.username}:${testOpenSearchDashboardsUrl.password}`
+            : undefined,
+        username: testOpenSearchDashboardsUrl.username ?? undefined,
+        password: testOpenSearchDashboardsUrl.password ?? undefined,
+        fullURL: testOpenSearchDashboardsUrl,
       };
     }
 
     const username =
       process.env.TEST_OPENSEARCH_DASHBOARDS_USERNAME || opensearchDashboardsTestUser.username;
     const password =
       process.env.TEST_OPENSEARCH_DASHBOARDS_PASSWORD || opensearchDashboardsTestUser.password;
+    const protocol = process.env.TEST_OPENSEARCH_DASHBOARDS_PROTOCOL || 'http';
+    const hostname = process.env.TEST_OPENSEARCH_DASHBOARDS_HOSTNAME || 'localhost';
+    const port = process.env.TEST_OPENSEARCH_DASHBOARDS_PORT
+      ? parseInt(process.env.TEST_OPENSEARCH_DASHBOARDS_PORT, 10)
+      : 5620;
+    const fullURL = new URL(`${protocol}://${username}:${password}@${hostname}:${port}`);
+
     return {
-      protocol: process.env.TEST_OPENSEARCH_DASHBOARDS_PROTOCOL || 'http',
-      hostname: process.env.TEST_OPENSEARCH_DASHBOARDS_HOSTNAME || 'localhost',
-      port: process.env.TEST_OPENSEARCH_DASHBOARDS_PORT
-        ? parseInt(process.env.TEST_OPENSEARCH_DASHBOARDS_PORT, 10)
-        : 5620,
+      protocol,
+      hostname,
+      port,
       auth: `${username}:${password}`,
       username,
       password,
+      fullURL,
```

Comment:
Should we add another property (maybe `serverUrl`?) that is the equivalent of `fullURL.toString().slice(0, -1)`? I'd much rather have it in one place here, with an explanatory comment rather than spread throughout the test files, where it appears as a somewhat mysterious string manipulation.

Reply:
That's a great suggestion. Will add another `serverUrl` property.